carbonmark-api/src/routes/users/post.ts

```diff
@@ -44,11 +47,23 @@ const handler = (fastify: FastifyInstance) =>
       .get();
 
     if (!userSnapshot.empty) {
-      return reply.code(403).send({
+      return reply.code(400).send({
         error: "A user with this handle is already registered!",
       });
     }
 
+    const isAuthenticated = verifyProfileSignature({
+      // first signature is without nonce
+      expectedAddress: createData.address,
+      signature: request.headers.authorization?.split(" ")[1] || "",
+    });
+
+    if (!isAuthenticated) {
+      return reply.status(403).send({
+        error: "Unauthorized: invalid signature.",
+      });
+    }
```

Comment:
This might be better handled in some middleware like the old `bearer.ts`, what do you think?

I'm all for not being religious about duplicated code but having the same logic like this in seperate files leaves the possibility open that the logic is changed in one place but not in all cases leading to lots of headaches.

Reply:
that... took wayyyyy longer than I care to admit... but I got it working!